src/inmanta/env.py

```diff
@@ -1094,14 +1094,41 @@ class VirtualEnv(ActiveEnv):
     Creates and uses a virtual environment for this process. This virtualenv inherits from the previously active one.
     """
 
+    _invalid_chars_in_path_re = re.compile(r"[\"$`]")
+
     def __init__(self, env_path: str) -> None:
         super(VirtualEnv, self).__init__(env_path=env_path)
+        self.validate_path(env_path)
         self.env_path: str = env_path
         self.virtual_python: Optional[str] = None
         self._using_venv: bool = False
         self._parent_python: Optional[str] = None
         self._path_pth_file = os.path.join(self.site_packages_dir, "inmanta-inherit-from-parent-venv.pth")
 
+    def validate_path(self, path: str) -> None:
+        """
+        Check for the presence of the following invalid characters in the given path.
+            - '$' because of a limitation in python venv causing prompt display issues:
+                - '$$' will be replaced by the pid of the process
+                - '${<var>}' will be expanded (possibly to empty string if var is undefined)
+
+            - '`' to prevent command substitution e.g. `whoami` would be interpreted in the prompt.
+
+            - '"' because of a limitation in python venv causing venv activation to fail silently for python<3.12 and
+              a crash for python>=3.12
```

Comment:
I think this docstring is missing a bit of context. The essential part of the problem is that this path is injected in the `activate` file of the created venv, which is a bash script. So we need to exclude all special characters that might be interpreted incorrectly in a bash string.

```
The given path is used in the `./bin/activate` file of the created venv without escaping any special characters. As such, we refuse all special characters here that might cause the given path to be interpreted incorrectly:

* $: Character used for variable expansion in bash strings.
* `: Character used to perform command substitution in bash strings.
* ": Character that will be interpreted incorrectly as the end of the string. 
```